client/aws/aws.go

```diff
@@ -44,23 +46,37 @@ const (
 )
 
 func NewClients(config Config) Clients {
+	s := newSession(config)
+	return newClients(s)
+}
+
+func NewClientsWithRole(config Config, roleARN string) Clients {
+	sess := newSession(config)
+
+	creds := stscreds.NewCredentials(sess, roleARN)
+
+	return newClients(sess, &aws.Config{Credentials: creds})
+}
+
+func newSession(config Config) *session.Session {
 	awsCfg := &aws.Config{
 		Credentials: credentials.NewStaticCredentials(config.AccessKeyID, config.AccessKeySecret, config.SessionToken),
 		Region:      aws.String(config.Region),
 	}
-	s := session.New(awsCfg)
-	clients := Clients{
-		AutoScaling:    autoscaling.New(s),
-		CloudFormation: cloudformation.New(s),
-		EC2:            ec2.New(s),
-		ELB:            elb.New(s),
-		IAM:            iam.New(s),
-		KMS:            kms.New(s),
-		Route53:        route53.New(s),
-		S3:             s3.New(s),
-	}
+	return session.New(awsCfg)
+}
```

Comment:
Nitpick: we tend to sort methods alphabetically. Public and private grouped.

Reply:
done